cmd/prepare-storage.go

```diff
@@ -114,90 +116,199 @@ func prepForInit(disks []string, sErrs []error, diskCount int) InitActions {
 		return WaitForHeal
+	} // No quorum wait for quorum number of disks.
+	return WaitForQuorum
+}
+
+type printFunc func(msg string)
+
+func newPrintFn() printFunc {
+	printed := false
+	return func(msg string) {
+		if !printed {
+			console.Println(msg)
+		}
+		printed = true
 	}
+}
 
-	// No Quorum.
-	if disksOffline >= quorum {
-		return WaitForQuorum
+func printHealMsg(storageDisks []StorageAPI, fn printFunc) {
+	msg := getHealMsg(storageDisks)
+	fn(msg)
+}
+
+func getHealMsg(storageDisks []StorageAPI) string {
+	msg := fmt.Sprintln("Data volume requires HEALING. Please run the following command:")
+	msg += "MINIO_ACCESS_KEY=%s "
+	msg += "MINIO_SECRET_KEY=%s "
+	msg += "minio control heal %s\n"
+	creds := serverConfig.GetCredential()
+	msg = fmt.Sprintf(msg, creds.AccessKeyID, creds.SecretAccessKey)
+	disksInfo, _, _ := getDisksInfo(storageDisks)
+	for i, info := range disksInfo {
+		msg += fmt.Sprintf(
+			"[%d/%d] %s - %s %s\n",
+			i+1,
+			len(storageDisks),
+			storageDisks[i],
+			humanize.IBytes(uint64(info.Total)),
+			func() string {
+				if info.Total > 0 {
+					return "online"
+				}
+				return "offline"
+			}(),
+		)
 	}
+	return msg
+}
 
-	// There is quorum or more corrupted disks, there is not enough good
-	// disks to reconstruct format.json.
-	if disksCorrupted >= quorum {
-		return Abort
+func printRegularMsg(storageDisks []StorageAPI, fn printFunc) {
+	msg := getRegularMsg(storageDisks)
+	fn(msg)
+}
+
+func getRegularMsg(storageDisks []StorageAPI) string {
+	msg := fmt.Sprintln(colorBlue("\nInitializing data volume."))
+	disksInfo, _, _ := getDisksInfo(storageDisks)
+	for i, info := range disksInfo {
+		msg += fmt.Sprintf(
+			"[%d/%d] %s - %s %s\n",
+			i+1,
+			len(storageDisks),
+			storageDisks[i],
+			humanize.IBytes(uint64(info.Total)),
+			func() string {
+				if info.Total > 0 {
+					return "online"
+				}
+				return "offline"
+			}(),
+		)
 	}
-	// Some of the formatted disks are possibly offline.
-	return WaitForHeal
+	return msg
 }
 
-func retryFormattingDisks(disks []string, storageDisks []StorageAPI) ([]StorageAPI, error) {
+func printFormatMsg(storageDisks []StorageAPI, fn printFunc) {
+	msg := getFormatMsg(storageDisks)
+	fn(msg)
+}
+
+func getFormatMsg(storageDisks []StorageAPI) string {
+	msg := fmt.Sprintln(colorBlue("\nInitializing data volume for the first time."))
+	disksInfo, _, _ := getDisksInfo(storageDisks)
+	for i, info := range disksInfo {
+		msg += fmt.Sprintf(
+			"[%d/%d] %s - %s %s\n",
+			i+1,
+			len(storageDisks),
+			storageDisks[i],
+			humanize.IBytes(uint64(info.Total)),
+			func() string {
+				if info.Total > 0 {
+					return "online"
+				}
+				return "offline"
+			}(),
+		)
+	}
+	return msg
+}
+
+func retryFormattingDisks(firstDisk bool, storageDisks []StorageAPI) error {
+	if storageDisks == nil {
+		return errInvalidArgument
+	}
 	nextBackoff := time.Duration(0)
 	var err error
-	done := false
+	var done bool
```

Comment:
The reason being consistent with the above `var err error` - 'done :=' is basically allocating we don't need to do that and let compiler have the default value based on the type. 

Since it is global to the loop this seemed better.